src/golang/cmd/server/handler/register_workflow.go

```diff
@@ -155,13 +155,18 @@ func (h *RegisterWorkflowHandler) Perform(ctx context.Context, interfaceArgs int
 		return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Unable to create workflow.")
 	}
 
-	execEnvByOpId, status, err := setupExecEnv(
+	txn, err := h.Database.BeginTx(ctx)
+	if err != nil {
+		return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Unable to create workflow.")
+	}
+	defer database.TxnRollbackIgnoreErr(ctx, txn)
+
+	execEnvByOpId, status, err := registerDependencies(
```

Comment:
@likawind Do we need to make a call to `setupCondaEnv` after this?

Reply:
Yes, the PRs are a bit messed up, and this call is included in https://github.com/aqueducthq/aqueduct/pull/1023/files

Reply:
Alternatively, I can put this PR: https://github.com/aqueducthq/aqueduct/pull/1026 against main so that you have one large PR to review, but the content will be more self-contained